doc/develop/west/zephyr-cmds.rst

```diff
@@ -281,3 +281,118 @@ Additional tips:
+        ...
+   ├── west.yml
+   └── zephyr
+       ├── module.yml
+       ├── patches
+       │   └── zephyr
+       │       └── my-zephyr-change.patch
+       └── patches.yml
+
+In this example, the ``west.yml`` file pins to a specific revision of Zephyr (``v4.1.0``).
+However, the application requires a change to main Zephyr project code in order to meet
+requirements.
+
+The ``patches.yml`` file can be used to keep metadata about the patch, to track its upstreaming
+status, and makes it very easy to remove patches that are no longer necessary. This makes it
+possible to maintain a tracked set of changes against specific revisions of zephyr modules
+without maintaining separate forks.
+
+.. code-block:: yaml
+  :alt: example of a ``patches.yml`` file
+
+    patches:
+      - path: zephyr/my-zephyr-change.patch
+        sha256sum: c676cd376a4d19dc95ac4e44e179c253853d422b758688a583bb55c3c9137035
+        module: zephyr
+        author: Obi-Wan Kenobi
+        email: obiwan@jedi.org
+        date: 2025-05-04
+        upstreamable: false
+        comments: |
+          An application-specific change we need for Zephyr.
+      - path: bootloader/mcuboot/my-tweak-for-mcuboot.patch
+        sha256sum: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+        module: zephyr
+        author: Obi-Wan Kenobi
+        email: obiwan@jedi.org
+        date: 2025-05-04
+        merge-pr: https://github.com/zephyrproject-rtos/zephyr/pull/<pr-number>
+        issue: https://github.com/zephyrproject-rtos/zephyr/issues/<issue-number>
+        merge-status: true
+        merge-commit: 1234567890abcdef1234567890abcdef12345678
+        merge-date: 2025-05-06
+        apply-command: git apply
+        comments: |
+          A change to mcuboot that has been merged already. We can remove this
+          patch when we update to the next release.
+
+Patches can easily be applied in an automated manner. For example:
+
+   .. code-block:: bash
+    :alt: a possibly automated workflow that uses west patch
+
+    west init -m <manifest repo> <workspace>
+    cd <workspace>
+    west update
+    west patch apply
+
+When it is time to update to a newer version of Zephyr, the ``west.yml`` file can be updated to
+point at e.g. ``v4.2.0``, and then the following commands can be run.
+
+   .. code-block:: bash
+    :alt: re-applying patches after a Zephyr update with west patch
+
+    west patch clean
+    west update
+    west patch apply -r # rolls-back all patches if one does not apply cleanly
+
+
+It is also possible to use ``west patch gh-fetch`` to fetch patches from a GitHub pull request and
+automatically create or update ``patches.yml`` file along with all local patch files.
+
+This can be useful when the author already has a number of changes captured in existing upstream
+pull requests
+
+.. code-block:: bash
+  :alt: fetching patches from GitHub pull request
+
+    west patch gh-fetch -o zephyrproject-rtos -r zephyr -pr <pr-number> -m zephyr -s
```

Comment:
Maybe use `--split-commits` in full as this is more descriptive as an example.

Reply:
Fixed! (used expanded arguments in general)